packages/cli/src/commands/start.ts

```diff
@@ -231,38 +228,32 @@ export class Start extends BaseCommand {
 	}
 
 	async initOrchestration() {
-		if (config.get('executions.mode') !== 'queue') return;
+		if (config.getEnv('executions.mode') !== 'queue') return;
+
+		// queue mode in single-main scenario
 
-		if (!config.get('leaderSelection.enabled')) {
-			await Container.get(SingleMainInstancePublisher).init();
+		if (!config.getEnv('multiMainSetup.enabled')) {
+			await Container.get(SingleMainSetup).init();
 			await Container.get(OrchestrationHandlerMainService).init();
 			return;
 		}
 
-		// multi-main scenario
-
-		const { MultiMainInstancePublisher } = await import(
-			'@/services/orchestration/main/MultiMainInstance.publisher.ee'
-		);
-
-		const multiMainInstancePublisher = Container.get(MultiMainInstancePublisher);
+		// queue mode in multi-main scenario
 
-		await multiMainInstancePublisher.init();
-
-		if (
-			multiMainInstancePublisher.isLeader &&
-			!Container.get(License).isMultipleMainInstancesLicensed()
-		) {
+		if (!Container.get(License).isMultipleMainInstancesLicensed()) {
 			throw new FeatureNotLicensedError(LICENSE_FEATURES.MULTIPLE_MAIN_INSTANCES);
 		}
 
 		await Container.get(OrchestrationHandlerMainService).init();
 
-		multiMainInstancePublisher.on('leadershipChange', async () => {
-			if (multiMainInstancePublisher.isLeader) {
+		const multiMainSetup = Container.get(MultiMainSetup);
+
+		await multiMainSetup.init();
+
+		multiMainSetup.on('leadershipChange', async () => {
+			if (multiMainSetup.isLeader) {
```

Comment:
Inside this `if` block, since this means that the current instance just became the leader, we should clear all previously existing `ActivationErrors` so that this instance would then report its own errors, if any happen, when initializing triggers wdyt?

Reply:
Addressed here: [7435dd1](https://github.com/n8n-io/n8n/pull/7643/commits/7435dd10135be87c898668cb1ead7f9633a5c7aa)